crates/ra_ide/src/syntax_highlighting.rs

```diff
@@ -406,6 +406,30 @@ fn highlight_element(
                 _ => h,
             }
         }
+        REF_EXPR => {
+            let ref_expr = element.into_node().and_then(ast::RefExpr::cast)?;
+            let expr = ref_expr.expr()?;
+            let field_expr = match expr {
+                ast::Expr::FieldExpr(fe) => fe,
+                _ => return None,
+            };
+
+            let expr = field_expr.expr()?;
+            let ty = match sema.type_of_expr(&expr) {
+                Some(ty) => ty,
+                None => {
+                    println!("No type :(");
+                    return None;
+                }
+            };
+            if !ty.is_packed(db) {
+                return None;
+            }
+
+            // FIXME account for alignment... somehow
```

Comment:
@matklad does RA currently track alignment in any way? The rust compiler currently uses this util fn: https://github.com/rust-lang/rust/blob/f3fadf6abd571868d70538561a0731ddd800003a/src/librustc_mir/util/alignment.rs#L7

Reply:
RA doesn't currently calculate the layout of any type.

Reply:
Should we highlight on all references to fields of packed structs, or should we not at all since that might cause confusion?

Reply:
Maybe we should highlight it differently as "maybe unsafe"?

Reply:
I'm not a fan of adding in a highlight class that might only exist temporarily. Is layout information something that would be provided by the chalk integration? If that's the case, maybe we can hold this off until then?

Reply:
> Is layout information something that would be provided by the chalk integration?

No, chalk only handles trait solving. It doesn't have anything to do with the layout computation.

Reply:
Should I just leave the FIXME in place for now? Or does RA never plan to account for this?

Reply:
It seems unlikely that we'll implement layout computation before librarification is far enough. But I do think it makes sense to have a comment there that explains this. (i.e. "this needs layout computation to be correct")

Reply:
Done